src/catalog/catalog_set.cpp

```diff
@@ -175,64 +171,68 @@ CatalogEntry* CatalogSet::dropEntryNoLock(const Transaction* transaction, const
 
 void CatalogSet::alterTableEntry(const Transaction* transaction,
     const binder::BoundAlterInfo& alterInfo) {
-    {
-        std::unique_lock lck{mtx};
-        // LCOV_EXCL_START
-        validateExistNoLock(transaction, alterInfo.tableName);
-        // LCOV_EXCL_STOP
-        auto entry = getEntryNoLock(transaction, alterInfo.tableName);
-        KU_ASSERT(entry->getType() == CatalogEntryType::NODE_TABLE_ENTRY ||
-                  entry->getType() == CatalogEntryType::REL_TABLE_ENTRY);
-        const auto tableEntry = entry->ptrCast<TableCatalogEntry>();
-        auto newEntry = tableEntry->alter(alterInfo);
-        newEntry->setTimestamp(transaction->getID());
-        newEntry->setOID(tableEntry->getOID());
-        if (alterInfo.alterType == AlterType::RENAME_TABLE) {
-            // We treat rename table as drop and create.
-            dropEntryNoLock(transaction, alterInfo.tableName, entry->getOID());
-            auto createdEntry = createEntryNoLock(transaction, std::move(newEntry));
-            if (transaction->shouldAppendToUndoBuffer()) {
-                transaction->pushAlterCatalogEntry(*this, *entry, alterInfo);
-                transaction->pushCreateDropCatalogEntry(*this, *createdEntry, isInternal(),
-                    true /* skipLoggingToWAL */);
-            }
-        } else {
-            emplaceNoLock(std::move(newEntry));
-            if (transaction->shouldAppendToUndoBuffer()) {
-                transaction->pushAlterCatalogEntry(*this, *entry, alterInfo);
-            }
+    std::unique_lock lck{mtx};
+    // LCOV_EXCL_START
+    validateExistNoLock(transaction, alterInfo.tableName);
+    // LCOV_EXCL_STOP
+    auto entry = getEntryNoLock(transaction, alterInfo.tableName);
+    KU_ASSERT(entry->getType() == CatalogEntryType::NODE_TABLE_ENTRY ||
+              entry->getType() == CatalogEntryType::REL_TABLE_ENTRY);
+    const auto tableEntry = entry->ptrCast<TableCatalogEntry>();
+    auto newEntry = tableEntry->alter(transaction->getID(), alterInfo);
+    switch (alterInfo.alterType) {
+    case AlterType::RENAME: {
+        // We treat rename table as drop and create.
+        dropEntryNoLock(transaction, alterInfo.tableName, entry->getOID());
+        auto createdEntry = createEntryNoLock(transaction, std::move(newEntry));
+        if (transaction->shouldAppendToUndoBuffer()) {
+            transaction->pushAlterCatalogEntry(*this, *entry, alterInfo);
+            transaction->pushCreateDropCatalogEntry(*this, *createdEntry, isInternal(),
+                true /* skipLoggingToWAL */);
+        }
+    } break;
+    default: {
```

Comment:
I would explicitly enumerate type here and make default KU_UNREACHABLE

Reply:
Sounds good. Let me refactor this in my next PR, which I'm doing refactoring already.